ome_zarr/bioformats2raw.py

```diff
@@ -0,0 +1,98 @@
+"""
+Spec definitions
+"""
+
+import logging
+import os
+import re
+import tempfile
+from xml.etree import ElementTree as ET
+
+import ome_types
+
+from ome_zarr.io import ZarrLocation
+from ome_zarr.reader import Node
+from ome_zarr.reader import Spec as Base
+
+__author__ = "Open Microscopy Environment (OME)"
+__copyright__ = "Open Microscopy Environment (OME)"
+__license__ = "BSD-2-Clause"
+
+_logger = logging.getLogger(__name__)
+
+
+class bioformats2raw(Base):
+    @staticmethod
+    def matches(zarr: ZarrLocation) -> bool:
+        layout = zarr.root_attrs.get("bioformats2raw.layout", None)
+        return layout == 3
+
+    def __init__(self, node: Node) -> None:
+        super().__init__(node)
+        try:
+            data = self.handle(node)
+            if data.plates:
+                _logger.info("Plates detected. Skipping implicit loading")
+            else:
+                for idx, image in enumerate(data.images):
+                    series = node.zarr.create(str(idx))
+                    assert series.exists(), f"{series} is missing"
+                    _logger.info("found %s", series)
+                    subnode = node.add(series)
+                    if subnode:
+                        subnode.metadata["ome-xml:index"] = idx
+                        subnode.metadata["ome-xml:image"] = image
```

Comment:
If you add the following here, then you don't need https://github.com/ome/napari-ome-zarr/pull/47/
```
subnode.metadata["name"] = image.name
```

Reply:
But then plugins would/could overwrite the main metadata, right? Which means it becomes a question of the order that the plugins run in. Would:

```
subnode.plugins["bioformats2raw"].metadata["name"]
```
be better?

Reply:
As probably transpired from https://github.com/ome/ome-zarr-py/pull/140, I am also unclear on the contract and expectation for the `node.metadata` field. My impression was that this API has been largely driven by the napari use-case.Reading this, it looks like another goal is to introduce some form of metadata consolidation so that a consumer does not have to go up and down the hierarchy to assemble pieces of metadata, is that correct?

If so, I agree what is missing is some form of namespace to differentiate the metadata injected at different levels in the hierarchy. Taking advantage of the fact this is a Python dictonary, another proposal would be to group all the metadata under a top-level key e.g. `subnode.metadata["bioformats2raw"] = {"index": idx, "image": image", "name", image.name}`

Reply:
> .Reading this, it looks like another goal is to introduce some form of metadata consolidation so that a consumer does not have to go up and down the hierarchy to assemble pieces of metadata, is that correct?

Maybe. But now that we have the json-schema, I could also see just attaching objects at the right node level to prevent re-reading the file. That's essentially what would happen if the ome-types `OME` object becomes part of the bioformats2raw plugin's public API.

Reply:
In my mind I might expect something like Josh's original idea or `subnode.metadata["ome.xml"] = {"index": idx, "image": image", "name", image.name}` since this is metadata coming from the `ome.xml`?

The reason I like `subnode.metadata["name"] = image.name` is that any consumer doesn't have to know about `bioformats2raw` or xml etc to have the correct image name.
But if it's just `napari-ome-zarr` and we already have https://github.com/ome/napari-ome-zarr/pull/47 then 👍

Reply:
> is that any consumer doesn't have to know about bioformats2raw or xml etc to have the correct image name.

This possibly gets back to the question of whether or not this bf2raw parsing is core or not. If it is, then you're probably right that we should just encode the rules for what "wins" directly. If this is a plugin, though, then how would we handle misbehavior in the plugins?

Reply:
What is a plugin here? If it's a plugin, then it's not installed by default and you get to choose if you want to add it?

Reply:
> This possibly gets back to the question of whether or not this bf2raw parsing is core or not

Trying to answer this, this is where I would draw the line: 

- if a specification is published in https://ngff.openmicroscopy.org/, I consider it as `core` (even if `transitional`) and my expectation is that this library should support it
- a contrario, any specification _not_ defined in the upstream specification should be seen as an extension and should  be handled by some plugin/third-party mechanism. This might drive the discussions on the extensibility of this library which is a good thing.